src/systems/a320_systems/src/pneumatic.rs

```diff
@@ -1,27 +1,3041 @@
+    fn read(&mut self, reader: &mut SimulatorReader) {
+        self.engine_1_state = reader.read(&self.engine_1_state_id);
+        self.engine_2_state = reader.read(&self.engine_2_state_id);
+    }
+}
+
+/// A struct to hold all the pack related components
+struct PackComplex {
+    pack_flow_valve_flow_rate_id: VariableIdentifier,
+    pack_container: PneumaticPipe,
+    exhaust: PneumaticExhaust,
+    pack_flow_valve: DefaultValve,
+    pack_flow_valve_controller: PackFlowValveController,
+}
+impl PackComplex {
+    fn new(context: &mut InitContext, engine_number: usize) -> Self {
+        Self {
+            pack_flow_valve_flow_rate_id: context
+                .get_identifier(format!("PNEU_PACK_{}_FLOW_VALVE_FLOW_RATE", engine_number)),
+            pack_container: PneumaticPipe::new(
+                Volume::new::<cubic_meter>(1.),
+                Pressure::new::<psi>(14.7),
+                ThermodynamicTemperature::new::<degree_celsius>(15.),
+            ),
+            exhaust: PneumaticExhaust::new(0.1),
+            pack_flow_valve: DefaultValve::new_closed(),
+            pack_flow_valve_controller: PackFlowValveController::new(context, engine_number),
+        }
+    }
+
+    fn update(&mut self, context: &UpdateContext, from: &mut impl PneumaticContainer) {
+        self.pack_flow_valve_controller
+            .update(context, self.pack_flow_valve.fluid_flow());
+
+        self.pack_flow_valve
+            .update_open_amount(&self.pack_flow_valve_controller);
+
+        self.pack_flow_valve
+            .update_move_fluid(context, from, &mut self.pack_container);
+
+        self.exhaust
+            .update_move_fluid(context, &mut self.pack_container);
+    }
+}
+impl PneumaticContainer for PackComplex {
+    fn pressure(&self) -> Pressure {
+        self.pack_container.pressure()
+    }
+
+    fn volume(&self) -> Volume {
+        self.pack_container.volume()
+    }
+
+    fn temperature(&self) -> ThermodynamicTemperature {
+        self.pack_container.temperature()
+    }
+
+    fn change_fluid_amount(&mut self, volume: Volume) {
+        self.pack_container.change_fluid_amount(volume);
+    }
+
+    fn update_temperature(&mut self, temperature_change: TemperatureInterval) {
+        self.pack_container.update_temperature(temperature_change);
+    }
+}
+impl SimulationElement for PackComplex {
+    fn accept<T: SimulationElementVisitor>(&mut self, visitor: &mut T) {
+        self.pack_flow_valve_controller.accept(visitor);
+
+        visitor.visit(self);
+    }
+
+    fn write(&self, writer: &mut SimulatorWriter) {
+        writer.write(
+            &self.pack_flow_valve_flow_rate_id,
+            self.pack_flow_valve.fluid_flow(),
+        );
+    }
+}
+
+// In the future, this will be done by the ACSC, hence why I have used an external controller and not the BMC
+struct PackFlowValveController {
+    pack_toggle_pb_id: VariableIdentifier,
+    pack_pb_is_auto: bool,
```

Comment:
**thought**: Should this be represented by an overhead panel type?

Reply:
Related to an earlier comment, the issue is that the local variables `AIRCOND_PACK{}_TOGGLE` do not follow the `...PB_IS_ON` and `...PB_HAS_FAULT` convention that's required to use the `AutoOffFaultPushButton`. I don't really feel comfortable renaming them in the whole project for now.